go.mod

```diff
@@ -12,6 +12,7 @@ require (
 	github.com/jessevdk/go-flags v1.6.1
 	github.com/onsi/ginkgo/v2 v2.27.2
 	github.com/onsi/gomega v1.38.2
+	go.yaml.in/yaml/v4 v4.0.0-rc.3
```

Comment:
may just need to run a `go mod tidy` here.

overall going to `go.yaml.in/yaml/v3` make sense to me